src/analytics.js

```diff
@@ -804,11 +762,86 @@ window.Altis.Analytics.registerAttribute = ( name, value ) => {
  * @param {string} name The metric name.
  * @param {*} value The metric value, can be a number, or callback or Promise that returns a number.
  */
-window.Altis.Analytics.registerMetric = ( name, value ) => {
+Altis.Analytics.registerMetric = ( name, value ) => {
 	_metrics[ name ] = value;
 	Analytics.updateAudiences();
 };
 
-// Fire a ready event once userland API has been exported.
-const readyEvent = new CustomEvent( 'altis.analytics.ready' );
-window.dispatchEvent( readyEvent );
+/**
+ * Start recording default events and trigger onReady event.
+ */
+function startAnalytics() {
+	// Avoid re-running everything.
+	if ( Altis.Analytics.Ready ) {
+		return;
+	}
+
+	// Set initial endpoint data.
+	Analytics.mergeEndpointData( Data.Endpoint || {} );
+
+	// Track sessions.
+	document.addEventListener( 'visibilitychange', () => {
+		if ( document.hidden ) {
+			// On hide increment elapsed time.
+			elapsed += Date.now() - start;
+			// Fire session stop event.
+			Analytics.record( '_session.stop' );
+		} else {
+			// On show reset start time.
+			start = Date.now();
+			// Reset subSessions.
+			subSessionId = uuid();
+			subSessionStart = Date.now();
+			// Fire session start event.
+			Analytics.record( '_session.start' );
+		}
+	} );
+
+	/**
+	 * Record the default page view.
+	 */
+	const recordPageView = () => {
+		// Session start.
+		Analytics.record( '_session.start' );
+		// Record page view event & create/update endpoint immediately.
+		Analytics.record( 'pageView', false );
+	};
+
+	if ( document.readyState === 'interactive' || document.readyState === 'complete' || document.readyState === 'loaded' ) {
+		recordPageView();
+	} else {
+		window.addEventListener( 'DOMContentLoaded', recordPageView );
+	}
+
+	// Flush remaining events.
+	window.addEventListener( 'beforeunload', () => {
+		Analytics.record( '_session.stop', false );
+	} );
+
+	// Fire a ready event once userland API has been exported.
+	Altis.Analytics.Ready = true;
+	const readyEvent = new CustomEvent( 'altis.analytics.ready' );
+	window.dispatchEvent( readyEvent );
+}
+
+// Check if this is anonymous or not by looking for endpoint user ID.
+const isAnonymous = ! ( Data.Endpoint && Data.Endpoint.User && Data.Endpoint.User.UserId );
+const consentType = 'statistics' + ( isAnonymous ? '-anonymous' : '' );
+
+// Check Altis Consent feature is in use.
+if ( Altis.Analytics.UseConsent ) {
+	// Check cookie directly for an early match.
+	if ( document.cookie.match( `_altis_consent_${ consentType }=allow` ) ) {
```

Comment:
Since prefix `_altis_consent_` is filterable [here](https://github.com/humanmade/altis-consent/blob/main/inc/namespace.php#L36), Can we use [Altis.Consent.getCookie](https://github.com/humanmade/altis-consent/blob/main/assets/js/main.js#L67) instead?

Reply:
This script is loaded really early so we can’t rely on that, but good call, we can probably use the same filter and pass it through.